AppCenterAnalytics/AppCenterAnalytics/MSEventProperties.m

```diff
@@ -0,0 +1,153 @@
+#import "MSEventProperties.h"
+#import "MSEventPropertiesInternal.h"
+#import "MSAnalyticsInternal.h"
+#import "MSBooleanTypedProperty.h"
+#import "MSConstants+Internal.h"
+#import "MSDateTimeTypedProperty.h"
+#import "MSDoubleTypedProperty.h"
+#import "MSLogger.h"
+#import "MSLongTypedProperty.h"
+#import "MSStringTypedProperty.h"
+
+static NSString *const kMSNullPropertyKeyMessage = @"Key cannot be null. Property will not be added.";
+static NSString *const kMSNullPropertyValueMessage = @"Value cannot be null. Property will not be added.";
+
+@implementation MSEventProperties
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    _properties = [NSMutableDictionary new];
+  }
+  return self;
+}
+
+- (instancetype)initWithDictionary:(NSDictionary<NSString *, NSString *> *)properties {
+  if ((self = [self init])) {
+    for (NSString *propertyKey in properties) {
+      MSStringTypedProperty *stringProperty = [MSStringTypedProperty new];
+      stringProperty.name = propertyKey;
+      stringProperty.value = properties[propertyKey];
+      _properties[propertyKey] = stringProperty;
+    }
+  }
+  return self;
+}
+
+#pragma mark - NSCoding
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [coder encodeObject:self.properties];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  if ((self = [self init])) {
+    [coder decodeObject];
+  }
+  return self;
+}
+
+#pragma mark - Public methods
+
+- (instancetype)setString:(NSString *)value forKey:(NSString *)key {
+  if (!key) {
+    MSLogWarning([MSAnalytics logTag], kMSNullPropertyKeyMessage);
+    return self;
+  }
+  if (!value) {
+    MSLogWarning([MSAnalytics logTag], kMSNullPropertyValueMessage);
+    return self;
+  }
+  MSStringTypedProperty *stringProperty = [MSStringTypedProperty new];
+  stringProperty.name = key;
+  stringProperty.value = value;
+  self.properties[key] = stringProperty;
+  return self;
+}
+
+- (instancetype)setDouble:(double)value forKey:(NSString *)key {
+  if (!key) {
+    MSLogWarning([MSAnalytics logTag], kMSNullPropertyKeyMessage);
+    return self;
+  }
+  if (value == (double)INFINITY || value == (double)NAN) {
+    MSLogError([MSAnalytics logTag], @"Double value for property '%@' must be finite (cannot be INFINITY or NAN).", key);
```

Comment:
Not consistent log level.

Reply:
Again, here actually android used error, we thought. Can double check.